src/main.rs

```diff
@@ -15,6 +15,17 @@ fn remove_non_printable(s: &str) -> String {
         .collect()
 }
 
+fn send(
+    serial_port: Option<&mut Box<dyn SerialPort>>,
+    buf: &[u8],
+) -> Result<(), Box<dyn std::error::Error>> {
+    if let Some(p) = serial_port {
+        p.write(buf)?;
```

Comment:
I just learned that today: use `write_all` if you want to write the whole buffer.